From efebb1f5b68372d517e2279159ca5b9eaac30265 Mon Sep 17 00:00:00 2001
From: Matt Benjamin <mbenjamin@redhat.com>
Date: Tue, 23 Aug 2016 14:18:43 -0400
Subject: [PATCH] librgw: add API version defines for librgw and rgw_file

This change borrows the major, minor+extra format used by libcephfs.
The version numbering is starting at 1,1,0 on the theory that the
implicit version at Jewel is 1,0,0.

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>
(cherry picked from commit 355ccc5ded7f0b459bca24bc8b504b41807c583d)
(cherry picked from commit 2a9233f1fbff38c12d378d5b6528a7e576a62e3e)

Cherry-picking to Fedora to move nfs-ganesha-rgw along, see
https://bugzilla.redhat.com/1388690
---
 src/include/rados/librgw.h   | 7 +++++++
 src/include/rados/rgw_file.h | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/src/include/rados/librgw.h b/src/include/rados/librgw.h
index 9ba8f3a..c20e96b 100644
--- a/src/include/rados/librgw.h
+++ b/src/include/rados/librgw.h
@@ -18,6 +18,13 @@
 extern "C" {
 #endif
 
+#define LIBRGW_VER_MAJOR 1
+#define LIBRGW_VER_MINOR 1
+#define LIBRGW_VER_EXTRA 0
+
+#define LIBRGW_VERSION(maj, min, extra) ((maj << 16) + (min << 8) + extra)
+#define LIBRGW_VERSION_CODE LIBRGW_VERSION(LIBRGW_VER_MAJOR, LIBRGW_VER_MINOR, LIBRGW_VER_EXTRA)
+
 typedef void* librgw_t;
 int librgw_create(librgw_t *rgw, int argc, char **argv);
 void librgw_shutdown(librgw_t rgw);
diff --git a/src/include/rados/rgw_file.h b/src/include/rados/rgw_file.h
index 4bb8a93..69995ca 100644
--- a/src/include/rados/rgw_file.h
+++ b/src/include/rados/rgw_file.h
@@ -24,6 +24,12 @@
 extern "C" {
 #endif
 
+#define LIBRGW_FILE_VER_MAJOR 1
+#define LIBRGW_FILE_VER_MINOR 1
+#define LIBRGW_FILE_VER_EXTRA 0
+
+#define LIBRGW_FILE_VERSION(maj, min, extra) ((maj << 16) + (min << 8) + extra)
+#define LIBRGW_FILE_VERSION_CODE LIBRGW_FILE_VERSION(LIBRGW_FILE_VER_MAJOR, LIBRGW_FILE_VER_MINOR, LIBRGW_FILE_VER_EXTRA)
 
 /*
  * object types
